src/app-layer-ntp.c

```diff
@@ -38,250 +38,9 @@
-                    proto_name, ALPROTO_NTP, 0, NTP_MIN_FRAME_LEN,
-                    NTPProbingParser, NULL)) {
-                SCLogDebug("No NTP app-layer configuration, enabling NTP"
-                    " detection UDP detection on port %s.",
-                    NTP_DEFAULT_PORT);
-                AppLayerProtoDetectPPRegister(IPPROTO_UDP,
-                    NTP_DEFAULT_PORT, ALPROTO_NTP, 0,
-                    NTP_MIN_FRAME_LEN, STREAM_TOSERVER,
-                    NTPProbingParser, NULL);
-            }
-
-        }
-
-    }
-
-    else {
-        SCLogDebug("Protocol detecter and parser disabled for NTP.");
-        return;
-    }
-
-    if (AppLayerParserConfParserEnabled("udp", proto_name)) {
-
-        SCLogDebug("Registering NTP protocol parser.");
-
-        /* Register functions for state allocation and freeing. A
-         * state is allocated for every new NTP flow. */
-        AppLayerParserRegisterStateFuncs(IPPROTO_UDP, ALPROTO_NTP,
-            NTPStateAlloc, NTPStateFree);
-
-        /* Register request parser for parsing frame from server to client. */
-        AppLayerParserRegisterParser(IPPROTO_UDP, ALPROTO_NTP,
-            STREAM_TOSERVER, RustNTPParseRequest);
-
-        /* Register response parser for parsing frames from server to client. */
-        AppLayerParserRegisterParser(IPPROTO_UDP, ALPROTO_NTP,
-            STREAM_TOCLIENT, RustNTPParseResponse);
-
-        /* Register a function to be called by the application layer
-         * when a transaction is to be freed. */
-        AppLayerParserRegisterTxFreeFunc(IPPROTO_UDP, ALPROTO_NTP,
-            NTPStateTxFree);
-
-        // AppLayerParserRegisterLoggerFuncs(IPPROTO_UDP, ALPROTO_NTP,
-        //     NTPGetTxLogged, NTPSetTxLogged);
-
-        /* Register a function to return the current transaction count. */
-        AppLayerParserRegisterGetTxCnt(IPPROTO_UDP, ALPROTO_NTP,
-            NTPGetTxCnt);
-
-        /* Transaction handling. */
-        AppLayerParserRegisterGetStateProgressCompletionStatus(ALPROTO_NTP,
-            NTPGetAlstateProgressCompletionStatus);
-        AppLayerParserRegisterGetStateProgressFunc(IPPROTO_UDP,
-            ALPROTO_NTP, NTPGetStateProgress);
-        AppLayerParserRegisterGetTx(IPPROTO_UDP, ALPROTO_NTP,
-            NTPGetTx);
-
-        /* Application layer event handling. */
-        AppLayerParserRegisterHasEventsFunc(IPPROTO_UDP, ALPROTO_NTP,
-            NTPHasEvents);
-
-        /* What is this being registered for? */
-        AppLayerParserRegisterDetectStateFuncs(IPPROTO_UDP, ALPROTO_NTP,
-            NULL, NTPGetTxDetectState, NTPSetTxDetectState);
-
-        AppLayerParserRegisterGetEventInfo(IPPROTO_UDP, ALPROTO_NTP,
-            NTPStateGetEventInfo);
-        AppLayerParserRegisterGetEventsFunc(IPPROTO_UDP, ALPROTO_NTP,
-            NTPGetEvents);
-    }
-    else {
-        SCLogDebug("NTP protocol parsing disabled.");
-    }
+    register_ntp_parser();
```

Comment:
should this use the rs_ prefix like other Rust code we call from C?

Reply:
Sure, that's a trivial change.
Do you prefer me to add one patch to this PR, or re-send another one?

Reply:
Lets wait to see if @jasonish comes up with other change requests. Normally I'd prefer a new PR,but it seems a bit excessive here :)